app/uk/gov/hmrc/helptosavefrontend/controllers/ReminderController.scala

```diff
@@ -0,0 +1,115 @@
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Either, Failure, Success}
+
+@Singleton
+class ReminderController @Inject() (val helpToSaveReminderService: HelpToSaveReminderService,
+                                    val helpToSaveService:         HelpToSaveService,
+                                    val sessionStore:              SessionStore,
+                                    val authConnector:             AuthConnector,
+                                    val metrics:                   Metrics,
+                                    val auditor:                   HTSAuditor,
+                                    cpd:                           CommonPlayDependencies,
+                                    mcc:                           MessagesControllerComponents,
+                                    errorHandler:                  ErrorHandler,
+                                    reminderFrequencySet:          reminder_frequency_set,
+                                    reminderConfirmation:          reminder_confirmation)(implicit val crypto: Crypto,
+                                                                                          implicit val transformer:                 NINOLogMessageTransformer,
+                                                                                          implicit val reminderFrequencyValidation: ReminderFrequencyValidation,
+                                                                                          val frontendAppConfig:                    FrontendAppConfig,
+                                                                                          val config:                               Configuration,
+                                                                                          val env:                                  Environment,
+                                                                                          ec:                                       ExecutionContext)
+
+  extends BaseController(cpd, mcc, errorHandler) with HelpToSaveAuth with SessionBehaviour with Logging {
+
+  private def backLink: String = routes.AccessAccountController.accessAccount().url
+
+  def getSelectRendersPage(): Action[AnyContent] = authorisedForHtsWithNINO { implicit request ⇒ implicit htsContext ⇒
+
+    Ok(reminderFrequencySet(ReminderForm.giveRemindersDetailsForm(), Some(backLink)))
+
+  }(loginContinueURL = routes.ReminderController.selectRemindersSubmit().url)
+
+  def selectRemindersSubmit(): Action[AnyContent] = authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
+
+    ReminderForm.giveRemindersDetailsForm().bindFromRequest().fold(
+      withErrors ⇒ {
+        Ok(reminderFrequencySet(withErrors))
+      },
+
+      success ⇒
+        {
+
+          //Get the user name
+          val userName = htsContext.userDetails match {
+            case Left(x)  ⇒ ""
```

Comment:
Are we expecting not to have a first name?  If not then we should fail and log in this case